lib/src/command/global_run.dart

```diff
@@ -20,14 +19,9 @@ class GlobalRunCommand extends PubCommand {
   String get invocation => "pub global run <package>:<executable> [args...]";
   bool get allowTrailingOptions => false;
 
-  /// The mode for barback transformers.
-  BarbackMode get mode => new BarbackMode(argResults["mode"]);
-
   GlobalRunCommand() {
     argParser.addFlag("checked",
         abbr: "c", help: "Enable runtime type checks and assertions.");
-    argParser.addOption("mode",
-        defaultsTo: "release", help: 'Mode to run transformers in.');
```

Comment:
If we're going to give `build` and `serve` helpful error messages, we might as well do so for `--mode` as well (also below).